pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -121,9 +118,10 @@ in stdenvNoCC.mkDerivation (args // {
     dotnet-sdk
   ];
 
-  makeWrapperArgs = args.makeWrapperArgs or [ ] ++ [
-    "--prefix LD_LIBRARY_PATH : ${dotnet-sdk.icu}/lib"
-  ];
+  makeWrapperArgs =
+    args.makeWrapperArgs or [ ] ++
+    lib.optional (dotnet-sdk ? icu)
+      "--prefix LD_LIBRARY_PATH : ${dotnet-sdk.icu}/lib";
```

Comment:
The source built sdk is patched to find libicu on the nix store path of the icu derivation it's built with.  It does the same thing with libkrb5.

Reply:
Yes, however, do packages compiled with `buildDotnetModule` have the correct ICU in their `rpath`?

I remember the compiler hardcoding the interpreter and rpath to something in the non-source builds

Reply:
As long as they use the runtime packages from the source-built sdk, they should just work, because all elfs are compiled in nix.  For ICU specifically the RPATH should no longer matter, because it's patched to find it in /nix/store.

Reply:
Yeah, I'm just mentioning this because even though we `patchelf`'d the pre-compiled `dotnet`, it was still setting the interpreter and rpath wrong (.NET 7 fixed that I don't know what they did exactly).

Reply:
And also keeping `passthrough.icu` will let us know which ICU a .NET SDK is using as 3.1 uses one and 6.0 uses another (and I assume it might change in the future as well).

Don't know why someone might need it but I don't see any issues in keeping it for compatibility with the other already existing .NET packages

Reply:
The way it's being used here is basically to decide if we even need to create a wrapper for outputs from dotnet.  Perhaps we could put something in the sdk to abstract whatever wrapping is required.